### ISSUE TITLE
oidc_settings dictionnary is not well formatted in the /etc/ood/config/ood_portal.yml

### ISSUE THREAD

**Issue by xpillons** [OSC/ood-ansible#115]
I'm using version 1.8 to setup Azure AD OIDC.
When trying to set the oidc variables in an override.yml file like that
```yml
oidc_remote_user_claim: upn
oidc_scope: 'openid profile email groups'
oidc_settings:
  OIDCPassIDTokenAs: serialized
  OIDCPassRefreshToken: On
  OIDCPassClaimsAs: environment
  OIDCStripCookies: 'mod_auth_openidc_session mod_auth_openidc_session_chunks mod_auth_openidc_session_0 mod_auth_openidc_session_1'
```
The settings are not well converted in proper yml format and the ansible playbook is returning this error
```
fatal: [ondemand]: FAILED! => {"changed": true, "cmd": "/opt/ood/ood-portal-generator/sbin/update_ood_portal --force", "delta": "0:00:00.353958", "end": "2021-07-02 14:02:28.243094", "msg": "non-zero return code", "rc": 1, "start": "2021-07-02 14:02:27.889136", "stderr": "(/etc/ood/config/ood_portal.yml): could not find expected ':' while scanning a simple key at line 364 column 1\nRun 'update_ood_portal --help' to see a full list of available options.", "stderr_lines": ["(/etc/ood/config/ood_portal.yml): could not find expected ':' while scanning a simple key at line 364 column 1", "Run 'update_ood_portal --help' to see a full list of available options."], "stdout": "", "stdout_lines": []}
```

Here is how the settings are written in the ood_portal.yml file
```
oidc_settings:
OIDCPassIDTokenAs serialized
OIDCPassRefreshToken True
OIDCPassClaimsAs environment
OIDCStripCookies 'mod_auth_openidc_session mod_auth_openidc_session_chunks mod_auth_openidc_session_0 mod_auth_openidc_session_1'
```

I've also noticed that the handler "update ood portal" is ignoring errors which shouldn't be the case, as here a malformed input should generate an error and not silently failed.

- How to define the oidc_settings so they are well formatted in the ppd_portal.yml file ?
- should the "update ood portal" failed be not ignored ?

**Comment by johrstrom** [comment]
Hey thanks for the ticket. This seems to be a bug. `update_ood_portal` returns error codes when things change. I guess we can capture the response code and make a decision off of that? But I believe it'll return 1 if things change, but everything went OK (I'd have to check on this logic).

For historic reasons, we allowed another way to define oidc settings, through `ood_auth_openidc` which will template out a separate oidc file in `conf.d`.  
https://github.com/OSC/ood-ansible#open-id-connect

**Comment by xpillons** [comment]
Thanks for the quick reply. Here the `update_ood_portal` is the one failing, so I guess this shouldn't be ignored.
I was not sure of which technic to use to set oidc, and before doing the automation I was tetsing manually, hence the path I took.

**Comment by xpillons** [comment]
Actually the fix seems simple. Use this block in the `templates/ood_portal.yml.j2`, note the added indent and the ':'

```
oidc_settings:
{% for key, value in oidc_settings.items() %}
  {{ key }}: {{ value | quote }}
{% endfor %}
{% endif %}
```

**Comment by johrstrom** [comment]
Hey sorry for the delay in releasing this. The fix is in 2.0.2 with a fix for #116 as well.
